source/fundamentals/auth.txt

```diff
@@ -0,0 +1,384 @@
+      ``MongoCredential`` class, use the
+      :java-docs:`createScramSha1Credental() </apidocs/mongodb-driver-core/com/mongodb/MongoCredential.html#createScramSha1Credential(java.lang.String,java.lang.String,char[])>`
+      method. Your code to instantiate a ``MongoClient`` should resemble the
+      following:
+
+      .. include:: /includes/fundamentals/code-snippets/auth-credentials-sha1.rst
+
+
+.. _mongodb-cr-auth-mechanism:
+
+``MONGODB-CR``
+~~~~~~~~~~~~~~
+
+``MONGODB-CR`` is a challenge-response authentication mechanism that uses your
+username and password to authenticate your user. This authentication
+mechanism was deprecated starting in MongoDB 3.6 and is no longer
+supported as of MongoDB 4.0.
+
+You cannot specify this method explicitly; refer to the fallback provided
+by the :ref:`Default authentication mechanism <default-auth-mechanism>` to
+connect using ``MONGODB-CR``.
+
+.. _mongodb-aws-auth-mechanism:
+
+``MONGODB-AWS``
+~~~~~~~~~~~~~~~
+
+.. note::
+   The MONGODB-AWS authentication mechanism is only available in MongoDB
+   versions 4.4 and later.
+
+The ``MONGODB-AWS`` authentication mechanism uses your Amazon Web Services
+Identity and Access Management (AWS IAM) credentials to authenticate your
+user.
+
+The following code snippets show how to specify the connection mechanism,
+using the following placeholders:
+
+* ``username`` - value of your ``AWS_ACCESS_KEY_ID``
+* ``password`` - value your ``AWS_SECRET_ACCESS_KEY``
+* ``hostname`` - network address of your MongoDB server, accessible by your client
+* ``port`` - port number of your MongoDB server
+* ``authenticationDb`` - MongoDB database that contains your user's
+  authentication data. If you omit this parameter, the driver uses the
+  default value ``admin``.
+* ``awsSessionToken`` - value your ``AWS_SESSION_TOKEN`` *(optional)*
+
+Select the :guilabel:`Connection String` or the :guilabel:`MongoCredential`
+tab below for instructions and sample code for specifying this authentication
+mechanism:
+
+.. tabs::
+
+   .. tab::
+      :tabid: Connection String
+
+      To specify the ``MONGODB-AWS`` authentication mechanism using a
+      connection string, pass the ``MONGODB-AWS`` value as a parameter
+      of ``authMechanism`` in your connection string. Your code to instantiate
+      a ``MongoClient`` should resemble the following:
+
+      .. code-block:: java
+
+         MongoClient mongoClient = MongoClients.create("mongodb://<username>:<password>@<hostname>:<port>/?authSource=<authenticationDb>&authMechanism=MONGODB-AWS");
+
+      If you need to specify an AWS session token, include it in the
+      ``authMechanismProperties`` parameter as follows:
+
+      .. code-block:: java
+
+         MongoClient mongoClient = MongoClients.create("mongodb://<username>:<password>@<hostname>:<port>/?authSource=<authenticationDb>&authMechanism=MONGODB-AWS&authMechanismProperties=AWS_SESSION_TOKEN:<awsSessionToken>");
```

Comment:
previous snippets have trained me to look for a key-value pairing (i would expect `AWS_SESSION_TOKEN=<token>`), do you think you can add an explicit mention that the ``authMechanismProperties``query parameter is used to declare and define the AWS_SESSION_TOKEN?

Reply:
I think the description prior to this mentions that they'll need to include the AWS session token in that property and the code example shows how it is formatted using the placeholders. Are you suggesting rewording it? Should this be in an admonition instead?

Reply:
Ah sorry, I didn't quite express myself clearly here. I meant to say that maybe we should mention that it's not merely a value for the `authMechanismProperties` parameter, but actually another way of expressing values. I could see a user pasting their `<awsSessionToken>` directly into the value of `authMechanismProperties` otherwise.

Reply:
Ok, will clarify.